posydon/popsyn/binarypopulation.py

```diff
@@ -235,13 +246,15 @@ def _safe_evolve(self, **kwargs):
 
         # Create temporary directory if it doesn't exist
         # Built to handle MPI
-        if self.comm is None:
+        if self.JOB_ID is None:
             if not os.path.exists(temp_directory):
                 os.makedirs(temp_directory)
         else:
-            if self.rank == 0:
-                if not os.path.exists(temp_directory):
-                    os.makedirs(temp_directory)
+            # Create a directory for MPI runs
+            try: 
+                os.makedirs(temp_directory)
+            except FileExistsError:
+                pass
```

Comment:
Why do we use a try clause here instead of checking, whether the directory already exists?

Reply:
Because the directory check can occur in multiple processes before one creates the directory. As such, it passes the check, but fails at the directory creation.

Reply:
Yes, it is useful to capture issues on creation. Why do we skip the check to avoid unnecessary exceptions?